lib/DB/TestUtils.pm

```diff
@@ -25,33 +27,52 @@ field starting with PARAM: and DATE: respectively.
 
 =cut
 
-sub buildHash ($input) {
+sub buildHash ($input, $config) {
 	my $output = {};
 	for my $key (keys %{$input}) {
 		if ($key =~ /^([A-Z_]+):(.*)/x) {
-			$output->{ lc($1) } = {} unless defined($output->{ lc($1) });
-			$output->{ lc($1) }->{$2} = $input->{$key} if defined($input->{$key});
+			my $field    = lc($1);
+			my $subfield = lc($2);
+			$output->{$field} = {} unless defined($output->{$field});
+			if ($key =~ /DATES:/) {
+				# Determine if each field is a date, a datetime or other (currently a boolean).
+				if (defined($input->{$key}) && $input->{$key} =~ /^\d{4}-\d{2}-\d{2}$/) {
+					my $dt = $strp_date->parse_datetime($input->{$key});
+					$output->{$field}->{$subfield} = $dt->epoch;
+				} elsif (defined($input->{$key}) && $input->{$key} =~ /^\d{4}-\d{2}-\d{2}T\d\d:\d\d:\d\dZ$/) {
+					my $dt = $strp_datetime->parse_datetime($input->{$key});
+					$output->{$field}->{$subfield} = $dt->epoch;
+				} elsif (grep { $_ eq $subfield } @{ $config->{param_boolean_fields} }) {
+					$output->{$field}->{$subfield} = int($input->{$key}) ? true : false if defined($input->{$key});
+				}
```

Comment:
This case should be removed from here, as I pointed out before.

Reply:
done.